### DISCUSSION TITLE
QUIC_BUFFER free after Send

### DISCUSSION THREAD

**Discussion by okbaswc** [microsoft/msquic#2289] [Q&A] [answered]
I'm malloc-ing the QUIC_BUFFER object, as well as the QUIC_BUFFER.Buffer object which is then sent over a stream. Does MSQUIC free this memory at all internally, or do I need to free it in QUIC_STREAM_EVENT_SEND_COMPLETE?

**Comment by nibanks** [comment, marked as answer]
You will have to free it in response to send complete.
